### PR TITLE
Start loading the plugins only when calling .ready, .inject or .listen

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/fastify/fastify/issues/491
Fixes https://github.com/fastify/fastify/pull/560

### CODE REVIEW

**Comment by mcollina** [comment]
done

**Comment by mcollina** [comment]
@jsumners updated. Sigh.

File: docs/Plugins-Guide.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ module.exports = function (fastify, options, next) {}
 ```
 Where `fastify` is (pretty obvious) the encapsulated Fastify instance, `options` is the options object and `next` is the function you **must** call when you plugin is ready.
 
-The Fastify's plugin model is fully reentrant and graph-based, it handles without any kind of problem asynchronous code and it guarantees the load order of the plugins, even the close order! *How?* Glad you asked, checkout [`avvio`](https://github.com/mcollina/avvio)!
+The Fastify's plugin model is fully reentrant and graph-based, it handles without any kind of problem asynchronous code and it guarantees the load order of the plugins, even the close order! *How?* Glad you asked, checkout [`avvio`](https://github.com/mcollina/avvio)! Fastify starts loading the plugin __after__ `.listen()`, `.inject()` or .ready()` are called.
```
Comment (jsumners) [inline docs/Plugins-Guide.md]: The leading `The` is unnecessary.

**Comment by mcollina** [comment]
@jsumners updated

**Comment by jsumners** [comment]
👍

**Pull request merged**